File: Hotelaria/src/main/java/com/dolts/hotelaria/models/ClienteModel.java
package com.dolts.hotelaria.models;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "cliente")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class ClienteModel implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String email;
    private String telefone;
    private String celular;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    private List<EnderecoModel> endereco;
    private LocalDateTime dataCadastro;
    private LocalDateTime dataUltimaAtualizacao;
    @Version
    private Long versao;
}

File: hotelaria-frontend/src/app/shared/models/cliente.model.ts
import {EnderecoModel} from "./endereco.model";

export class ClienteModel {
    id: number;
    email: string;
    telefone: string;
    celular: string;
    endereco: EnderecoModel;
    dataCadastro: Date;
    dataUltimaAtualizacao: Date;
    versao: number;
}
